online/src/app/product-details/product-details.component.html

```diff
@@ -8,8 +8,9 @@ <h1>Product: {{product.name}}</h1>
             <div>Description: {{product.description}}</div>
             <div>Price: {{product.price}}</div>
         </div>
-        <button class="remove" (click)="deleteProduct(product.id)" *ngIf="hasAdminRole">DELETE</button>
-        <button class="edit" [routerLink]="['/edit-product', product.id]" *ngIf="hasAdminRole">EDIT</button>
+        <br>
+        <button mat-raised-button class="remove" (click)="deleteProduct()" *ngIf="hasAdminRole">DELETE</button>
```

Comment:
In a real world scenario I would advise you to only add the ngIf once, if it repeats. I.e., put those under a div and the ngIf on that div.

Reply:
Really useful trick to remove duplicate code. Thanks for the advice. I moved them into a separate div and used the ngIf only once.